components/booking/pages/BookingPage.tsx

```diff
@@ -115,33 +127,49 @@ const BookingPage = (props: any): JSX.Element => {
         jitsu.track(telemetryEventTypes.bookingConfirmed, collectPageParameters())
       );
 
-      /*const res = await */ fetch("/api/book/event", {
-        body: JSON.stringify(payload),
-        headers: {
-          "Content-Type": "application/json",
-        },
-        method: "POST",
+      const content = await createBooking(payload).catch((e) => {
+        console.error(e.message);
+        setLoading(false);
+        setError(true);
```

Comment:
state would be better be represented as a string union


```ts
const [state, setState] = useState<'error' | 'loading' | 'ok'>('loading')
````


https://gist.github.com/KATT/98f10e7d451b05c09378b2df589cd0fd#explicit-state